docs/default.nix

```
@@ -1,23 +1,25 @@
 { mdbook
 , modules-docs
 , stdenv
+, lib
 }:
-let
-  source = import ../nix/source.nix;
-in
+with lib;
 stdenv.mkDerivation {
   name = "devshell-docs";
   buildInputs = [ mdbook ];
-  src = source.filter {
-    path = ./.;
-    allow = [
-      ./book.toml
-      (source.matchExt "md")
-    ];
-  };
+  src =
+    let fs = lib.fileset; in
+    fs.toSource {
+      root = ./.;
+      fileset = fs.unions [
+        (fs.fileFilter (file: file.hasExt "md") ./src)
+        (fs.fileFilter (file: true) ./theme)
```

Oh just randomly seeing this: There's no need to use `fileFilter` when you want to include every file of a directory, `./theme` by itself is also coerced to a file set to contain all its files :)

```suggestion
        ./theme
```